infra/templates/dns/template.yaml

```diff
@@ -0,0 +1,63 @@
+AWSTemplateFormatVersion: "2010-09-09"
+
+Description: >-
+  Creates the necessary components to manage DNS in the Async ID Check journey including logging and alternative record names.
+Parameters:
+  Environment:
+    Description: The name of the environment to deploy to
+    Type: String
+    AllowedValues:
+      - dev
+      - build
+      - staging
+      - integration
+      - production
+
+Conditions:
+  IsProduction: !Equals [ !Ref Environment, production]
+
+Resources:
+  PublicHostedZone:
+    Type: AWS::Route53::HostedZone
+    DeletionPolicy: Retain
+    UpdateReplacePolicy: Retain
+    Properties:
+      Name: !If
+        - IsProduction
+        - review-b-async.account.gov.uk
```

Comment:
Can we check with @sandy-may  @Pranjal-GDS about naming? 
I've now started to see `id-check-v2` being used alongside async so i'm worried about more fractured naming.

Reply:
This is the name I was instructed to use by @sandy-may

Reply:
Sounds good. I'm sure it is correct for the domains, but as i was introduced to this new name recently I'm also keen to flag it to the wider team about what names are in use for our team and product.

Reply:
I'm double checking on our product name in ID Check

ID Check V2 is the new production/app name for the entirety of mobile.

Reply:
ID check V2 has gone as a product name - I take it all back.

id-check-async is the way to go!